### PR TITLE
Group integration

### PR DESCRIPTION
**Pull request by D0nPiano**

### CODE REVIEW

**Comment by D0nPiano** [comment]
test !#/group/:id to see the groupDetail page with pickups, members and stores

File: client/app/components/_pickupList/pickupList.spec.js [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      $httpBackend.when("/api/auth/status/").respond(authData);
+      $httpBackend.when("/api/pickup-dates/?store=9").respond(pickupData);
+      $httpBackend.when("/api/stores/9/").respond(storeData);
+      
+      controller.userId = 1;
+      controller.addPickuplistInfos(pickupData);
+      let updatedData = controller.allPickups;
+      expect(updatedData[0].store).to.eventually.deep.equal(storeData);
+    });
+  });
```
Comment (Xbaal) [inline client/app/components/_pickupList/pickupList.spec.js]: I just had a look at this test, it seems really messy.
addPickuplistInfos(pickupData) gets as wel called by updatePickups(), which gets called once the Authentication is finished.
But this means, that the entries of controller.allPickups will already have promises in their .store attribute.
I am on this, i will comment here later.

Comment (Xbaal) [inline client/app/components/_pickupList/pickupList.spec.js]: btw i might be telling bs, i have been working for to long ;)

Comment (Xbaal) [inline client/app/components/_pickupList/pickupList.spec.js]: I got this working now, i have to figure out, whats the best way to contribute it here. Should i simply push?

Comment (D0nPiano) [inline client/app/components/_pickupList/pickupList.spec.js]: Simply pushing sounds good! Thank you for all your input! Going to see what i get done tonight.

File: client/app/components/_pickupList/pickupListItem/pickupListItem.styl [hunk 2]
```diff
@@ -0,0 +1,54 @@
+a
+    color: inherit
+    text-decoration: none
+    margin-left: .5em
+
+img
+    width: 2em
+    height: 2em
+    border-radius: 50%
```
Comment (nicksellen) [inline client/app/components/_pickupList/pickupListItem/pickupListItem.styl]: For that, we would need something like https://github.com/css-modules/css-modules (I tried it out in a toy project and it worked nicely). We could look into it in the future, not sure how it integrates with angular.

Comment (D0nPiano) [inline client/app/components/_pickupList/pickupListItem/pickupListItem.styl]: Ah, okay. Well, till then I'll just prefix the used classes with the component name then...

**Comment by D0nPiano** [comment]
So, the TODO that are left:

- [ ] make pickuplist tests prettier
- [ ] remove unnecessary comments
- [ ] add more useful groupDetail tests

File: client/app/components/_pickupList/pickupList.spec.js [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+  let { module } = angular.mock;
+
+  beforeEach(module(PickupListModule));
+  beforeEach(module(UserModule));
+  beforeEach(module(StoreModule));
+  beforeEach(module(AuthenticationModule));
+  beforeEach(module(PickupDateModule));
+
+  beforeEach(inject(($injector) => {
+    $httpBackend = $injector.get("$httpBackend");
```
Comment (tiltec) [inline client/app/components/_pickupList/pickupList.spec.js]: Each time you use $httpBackend, check if there are no outstanding & requests:
```
  afterEach(() => {
    $httpBackend.verifyNoOutstandingExpectation();
    $httpBackend.verifyNoOutstandingRequest();
  });
```
I'll add this here :)

**Comment by tiltec** [comment]
Great work so far integrating this into Angular structure @D0nPiano!

I saw that opening `http://localhost:3000/#!/group/1` fires 5 requests to `/api/stores/2/` directly after each other.. I'll try to figure out why that happens and find a solution.

File: client/app/components/_pickupList/pickupList.controller.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+     * - isFull
+     * - store (if showDetail == store)
+     */
+  addPickuplistInfos(pickups) {
+    angular.forEach(pickups, (currentPickup) => {
+      currentPickup.isUserMember = currentPickup.collector_ids.indexOf(this.userId) !== -1;
+      currentPickup.isFull = !(currentPickup.collector_ids.length < currentPickup.max_collectors);
+
+      if (this.showDetail === "store") {
+        currentPickup.store = this.Store.getById(currentPickup.store);
```
Comment (tiltec) [inline client/app/components/_pickupList/pickupList.controller.js]: I think overwrite a value with a promise is bad style. Would it be ok to write this?
```
this.Store.getById(currentPickup.store).then((store) => currentPickup.store = store)
```
does it need a closure?

**Comment by tiltec** [comment]
Where is `showDetail: date` used? could only find the "store" option.

File: client/app/components/_pickupList/pickupList.controller.js (hunk 4 above)
Comment (D0nPiano) [inline client/app/components/_pickupList/pickupList.controller.js]: Yes, that sounds better actually!

**Comment by D0nPiano** [comment]
@tiltec showDetail: "date" is more like a implicit default behavior. So if you didn't type in "store", it's always going to be "date" currently, wherefor you  can't see it in the code.

**Comment by alangecker** [comment]
integrated the group selection into the topbar/logo.
In my opinion it's much more chicer, especially 'cause most people are never in multiple communities.
groupselector
groupselector2

for clarification: is there a difference between 'groups' and 'communities'?

**Comment by D0nPiano** [comment]
I think I kind of lost track of this branch - what else needs to be done to merge this into master? I think all the todos I could think of are done now.

**Comment by tiltec** [comment]
> showDetail: "date" is more like a implicit default behavior. So if you didn't type in "store", it's always going to be "date" currently, wherefor you can't see it in the code.

PickupList is only used in GroupDetail so far, with the `showDetail: "store"` - did you inted the "date" default for future use, maybe inside a storeDetail page?

**Comment by tiltec** [comment]
I would propose to make the commit of @alangecker into a new pull request, so that we can merge this one.
Regarding the TODO: I will spend some time right now to reduce the number of calls to `/api/stores/2/`.

**Pull request merged**